### PR TITLE
[ENH] Expand contributing guidelines 🥳

### LINKED ISSUE #171: Add more instructions on contributing
### Description

- [x] Mention pre-commit for code formatting
- [x] Mention repo-specific dev environment setups
- [x] Before opening a PR, new contributors should (1) open an issue or (2) comment on the issue they are interested in working on
- [x] Can look for `quick fix`, `good first issue`, `pr welcome` labels
- [x] Request review from core maintainer
- [x] Change page name in index
- [x] Change .md file name

### Relevant documentation pages

https://neurobagel.org/contributing/pull_requests/

### Additional details

Nipreps has this cool little script to generate a grid of organization member GH profile icons in their documentation, using calls to the GH API 💡 : 
https://github.com/nipreps/nipreps.github.io/blob/d269b7d71930ff330ccacbc90ce3150c46430bc8/scripts/generate_members_gridpy (see it in action at https://www.nipreps.org/community/#current-members-of-the-github-organization).

Perhaps we could adapt the script to generate a similar grid in our docs for members of @neurobagel/dev and Neurobagel organization members? This way, it should also be updated automatically when the core maintainers change. This would replace our hardcoded team grid https://neurobagel.org/contributing/team/#developers

See also https://docs.github.com/en/rest/teams/members?apiVersion=2022-11-28

### PR DESCRIPTION
**Pull request by alyssadai**
Closes #171 

Changes proposed in this pull request:

- New sections:
  - "Identifying an issue to work on"
  - "Making a change"
  - "Have a question about contributing?"

- Updated sections:
  - "Pull request guidelines"
  - "Pull request reviews"

- `pull_requests.md` -> `CONTRIBUTING.md`

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Checks pass

### CODE REVIEW

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:100]: localhost seems to sneak into your links sometimes. copy/paste problem maybe?

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:6]: Want to add a short bit on "You will need to make a GH account" here. I know that's basic, but I like the idea of making this a really low first bar. 

c.f. the very well written tedana contribution doc: https://github.com/ME-ICA/tedana/blob/main/CONTRIBUTING.md

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:10]: Maybe instead: look for "good first issue" labels?

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:12]:
```suggestion
When you are ready to contribute, the best way to join the conversation is by opening an issue to describe a change you would like to see or would like to contribute. Opening an issue before you start contributing code is important because it allows others in the project to discuss your idea and tell you where your contribution would be the most helpful.```

Something like that?

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:16]:
```suggestion
    - If someone is already [assigned to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) this means that the problem is actively being worked on and a solution will soon be proposed. Feel free to share some helpful resources or pointers that may be interesting to the person who is working the issue - and check back in a couple of days.
```

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:24]: should we mention GH discussions as well? Do we have those enabled anywhere? Maybe one central discussion for very generic questions?

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:52]:
```suggestion
Refer to the README of the neurobagel repository you are contributing to for instructions on how to set up a development environment so you can test any local code changes you make.
```

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:60]:
```suggestion
Most repositories use tools to do automatic code formatting and linting according to the project's code style, 
```

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:61]:
```suggestion
which is pre-configured in the development environment for these repositories using [pre-commit](https://pre-commit.com/). Please make sure to set up these tools before you begin your work to ensure your contribution matches the existing code.
```

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:66]: it'd be `npx husky init` for the JS codebase. But that's in the README as well. @rmanaem wdyt: add that here as well or leave up to the repo?

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:111]:
```suggestion
A maintainer will review each PR and may provide comments or suggestions for the PR author to address.
```

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [inline docs/contributing/CONTRIBUTING.md:124]: I know that's how we explain it so far, but what about just saying: "if you have write permission" vs "if you don't" instead? It's more accurate, and it sounds a little less like a big deal.

**Review by surchs — COMMENTED** [review]
Thanks a lot @alyssadai, this is a big piece of work and really important to have a nice, welcoming first page for newcomers.

I left some comments with suggestions - take a look. 

Would it also make sense to link this prominently on our welcome page?

File: docs/contributing/CONTRIBUTING.md
Comment (alyssadai) [reply docs/contributing/CONTRIBUTING.md:100]: > localhost seems to sneak into your links sometimes. copy/paste problem maybe?

Heh, yep. I sometimes fill URLs in at the end, maybe too speedily. Looks like I missed the markdown link check here - will make sure to double check next time.

> more generally: do we want that? i.e. people requesting review directly? We should maybe add a bot to label and move the PR to the board. But it'd be tricky for external folks to know who to add here the page you linked includes many folks - and they may also be busy.

Yeah, good point. I'm okay with removing that recommendation for now until we have a better process. Sounds good?

File: docs/contributing/CONTRIBUTING.md
Comment (alyssadai) [reply docs/contributing/CONTRIBUTING.md:6]: Sure. I actually have seen that contribution template haha - was trying to strike a balance between avoiding having our version be too verbose (most people probably won't read) and setting the bar for contributing too high.

In this case, I think adding the tip about the GH account makes sense.

File: docs/contributing/CONTRIBUTING.md
Comment (rmanaem) [reply docs/contributing/CONTRIBUTING.md:66]: I think it's better to leave it up to the repo.

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [reply docs/contributing/CONTRIBUTING.md:10]: agree. I mainly meant: keeping this generic across repos seems clearer to me. But :cherries:

File: docs/contributing/CONTRIBUTING.md
Comment (rmanaem) [inline docs/contributing/CONTRIBUTING.md:128]: I think it's a good idea to link the team page on at least of the `core maintainers`.

File: docs/contributing/CONTRIBUTING.md
Comment (rmanaem) [reply docs/contributing/CONTRIBUTING.md:100]: > more generally: do we want that? i.e. people requesting review directly? We should maybe add a bot to label and move the PR to the board. But it'd be tricky for external folks to know who to add here the page you linked includes many folks - and they may also be busy.

I second this.
We could use the `_community` label for now and update its description to not be a `BOT ONLY` or add another label for bot PRs.

File: docs/contributing/CONTRIBUTING.md
Comment (rmanaem) [inline docs/contributing/CONTRIBUTING.md:84]: I think it's a good idea to mention "The checks in the template should remain intact as they're for the maintainer to go over during review."

File: docs/contributing/CONTRIBUTING.md
Comment (alyssadai) [reply docs/contributing/CONTRIBUTING.md:100]: That solution sounds good @neurobagel/dev.

I'll separate the labelling business into a separate issue, and remove the recommendation for requesting a review for now. We can update this page once the labels are updated.

File: docs/contributing/CONTRIBUTING.md
Comment (alyssadai) [reply docs/contributing/CONTRIBUTING.md:95]: Wdyt about making this more visible in the PR template itself? I'm hoping to avoid having duplication of info where possible, in case we change how we use the PR template in the future.

As part of https://github.com/neurobagel/.github/issues/9, I'm thinking to add a non-commented note to our [checklist](https://github.com/neurobagel/.github/blob/main/.github/pull_request_template.md#checklist) reminding PR authors to leave the checkboxes for the PR reviewer.

**Comment by alyssadai** [comment]
Thanks for your feedback @rmanaem and @surchs! 🥳

Have another look at the changes and the remaining conversations to see if you agree with the points made.

> Would it also make sense to link this prominently on our welcome page?

Sure - under the "Our team" section at the bottom perhaps?

File: docs/contributing/CONTRIBUTING.md
Comment (rmanaem) [reply docs/contributing/CONTRIBUTING.md:95]: Good call.

**Review by rmanaem — APPROVED** [review]
🧑‍🍳

File: docs/contributing/CONTRIBUTING.md
Comment (alyssadai) [reply docs/contributing/CONTRIBUTING.md]: See:
- https://github.com/orgs/neurobagel/projects/1/views/34?pane=issue&itemId=39843347

**Review by surchs — APPROVED** [review]
looks good @alyssadai, thanks for writing all this!

Re linking to our github page: 

> Sure - under the "Our team" section at the bottom perhaps?

I meant this doc: https://github.com/neurobagel/.github/blob/main/profile/README.md
not our website/documentation. i.e. the little text you see first when you go to https://github.com/neurobagel

File: docs/contributing/CONTRIBUTING.md
Comment (surchs) [reply docs/contributing/CONTRIBUTING.md:31]: OK, I agree!

**Comment by alyssadai** [comment]
> I meant this doc: https://github.com/neurobagel/.github/blob/main/profile/README.md
not our website/documentation. i.e. the little text you see first when you go to https://github.com/neurobagel

@surchs I see! No problem. Let's address that as part of this issue:
- https://github.com/neurobagel/.github/issues/9

**Pull request merged**
